codec-mqtt/src/main/java/io/netty/handler/codec/mqtt/MqttDecoder.java

```diff
@@ -464,6 +634,36 @@ private static MqttFixedHeader decodeFixedHeader(ByteBuf buffer) {
         return new Result<Integer>(result, numberOfBytesConsumed);
     }
 
+    private static Result<Integer> decodeVariableByteInteger(ByteBuf buffer) {
+        int remainingLength = 0;
+        int multiplier = 1;
+        short digit;
+        int loops = 0;
+        do {
+            digit = buffer.readUnsignedByte();
+            remainingLength += (digit & 127) * multiplier;
+            multiplier *= 128;
+            loops++;
+        } while ((digit & 128) != 0 && loops < 4);
+
+        // MQTT protocol limits Remaining Length to 4 bytes
+        if (loops == 4 && (digit & 128) != 0) {
+            return null;
+        }
+        return new Result<Integer>(remainingLength, loops);
+    }
+
+    private static Result<Integer> decode4bytesInteger(ByteBuf buffer) {
+        short msb = buffer.readUnsignedByte();
+        short secondByte = buffer.readUnsignedByte();
+        short thirdByte = buffer.readUnsignedByte();
+        short lsbSize = buffer.readUnsignedByte();
+        final int numberOfBytesConsumed = 4;
+        int result = msb << 24 | secondByte << 16 | thirdByte << 8 | lsbSize;
```

Comment:
Isn't that the same as calling `buffer.readInt(...)` ?

Reply:
Right, removed the duplicate code